templates/types/streaming/nextjs/app/api/chat/route.ts

```diff
@@ -62,14 +79,14 @@ export async function POST(request: NextRequest) {
     });
 
```

Comment:
@thucpn currently, TS doesn't have an event for starting the retrieval process (I will add it later), but can you send first a 

`appendEventData(vercelStreamData, 'Retrieving context for query ${query_str}');`

and change the text above to

`appendEventData(vercelStreamData, 'Retrieved ${nodes} sources to use as context for the query');`

Also please make sure that your UI supports displaying multiple events